.github/workflows/dotnet-format.yml

```diff
@@ -0,0 +1,75 @@
+﻿name: dotnet-format
+
+on:
+  pull_request:
+    branches: ["main", "feature*"]
+    paths:
+      - "webapi/**"
+      - "tools/**"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-format:
+    runs-on: ubuntu-latest
+    env:
+      NUGET_CERT_REVOCATION_MODE: offline
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          clean: true
+
+      - name: Get changed files
+        id: changed-files
+        if: github.event_name == 'pull_request'
+        uses: jitterbit/get-changed-files@v1
+        continue-on-error: true
+
+      - name: No C# files changed
+        id: no-csharp
+        if: github.event_name == 'pull_request' && steps.changed-files.outputs.added_modified == ''
+        run: echo "No C# files changed"
+
+      # This step will loop over the changed files and find the nearest .csproj file for each one, then store the unique csproj files in a variable
+      - name: Find csproj files
+        id: find-csproj
+        if: github.event_name != 'pull_request' || steps.changed-files.outputs.added_modified != '' || steps.changed-files.outcome == 'failure'
```

Comment:
I don't understand this condition...

We only run the following code if we DON'T have a PR or we have NO modified files OR we have a failure???

Is this correct?  Maybe add a comment?

Reply:
Good catch!